x-pack/plugins/upgrade_assistant/public/plugin.ts

```diff
@@ -53,6 +53,10 @@ export class UpgradeAssistantUIPlugin
         const appDependencies: AppDependencies = {
           kibanaVersionInfo,
           isReadOnlyMode: readonly,
+          // Infra plugin doesnt export anything as a public interface. So the only
+          // way we have at this stage for checking if the plugin is available or not
+          // is by checking if the startServices has the `infra` key.
+          isInfraPluginAvailable: plugins.hasOwnProperty('infra'),
```

Comment:
Thanks for adding this comment! How about we expose an `infra` property on the `plugins` object to make the interface more consistent?

```ts
// line 60
plugins: {
  // Infra plugin doesnt export anything as a public interface. So the only
  // way we have at this stage for checking if the plugin is available or not
  // is by checking if the startServices has the `infra` key.
  infra: plugins.hasOwnProperty('infra') ? {} : undefined,
  cloud,
  share,
  /* ... */
}
```

Reply:
Good point! I've addressed this with https://github.com/elastic/kibana/pull/111960/commits/9f5325da2d5b4130e51b8b9fbcb0343e33aa79ae